todo_app/data/trello_items.py

```diff
@@ -1,111 +1,134 @@
+        """
+        def parse_column(column):
+            return [Item.from_trello_card(card, column) for card in column["cards"]]
+        
+        response = self.trello_request("GET", f"/1/boards/{self.board_id}/lists", {"cards": "open"})
+
+        return sorted(list(itertools.chain(*[parse_column(col) for col in response])), key=lambda item: item.id)
+
+    def get_item(self, id):
+        """
+        Fetches the saved item with the specified ID.
+
+        Args:
+            id: The ID of the item.
+
+        Returns:
+            item: The saved item, or None if no items match the specified ID.
+        """
+        items = self.get_items()
+        return next((item for item in items if item.id == id), None)
+
+    def get_lists(self):
+        """
+        Gets all the lists for the trello board
+
+        Returns:
+            lists: a list of List objects
+        """
+        response = self.trello_request("GET", f"/1/boards/{self.board_id}/lists")
+        return [List.from_trello_list(trello_list) for trello_list in response]
+
+    def get_list_id_from_name(self, status_name: Status):
+        """
+        Gets the trello list id of a list with the given name, or None if it does not exist
+        
+        Args:
+            name: The name of the list.
+
+        Returns:
+            listId: the id of the list, or None if it does not exist 
+        """
+        trello_lists = self.get_lists()
+        return next((trello_list.id for trello_list in trello_lists if trello_list.name ==status_name.value), None)
+
+    def add_item(self, title):
+        """
+        Adds a new item with the specified title to the session.
+
+        Args:
+            title: The title of the item.
+
+        Returns:
+            item: The saved item.
+        """
+
+        todo_id = self.get_list_id_from_name(Status.ToDo)
+        self.trello_request("POST", f"/1/cards", {"idList": todo_id, "name": title})
+
+    def progress_item(self, id):
+        """
+        Progresses an existing item one step (To do -> Doing -> Done). If no existing item matches the ID of the specified item, nothing is marked as done
+
+        Args:
+            id: The ID of the item to progress 
+        """
+        next_status = Status.get_next(self.get_item(id).status)
+        status_id = self.get_list_id_from_name(next_status)
+        self.trello_request("PUT", f"/1/cards/{id}", {"idList": status_id})
+
+    def remove_item(self, id):
+        """
+        Removes an existing item from the board. If no existing item matches the ID of the specified item, no item is removed
+
+        Args:
+            id: The ID of the item to remove
+        """
+        self.trello_request("DELETE", f"/1/cards/{id}")
+
+
+    def create_board(self, name):
+        """
+        Creates a new board in trello
+        """
+
+        response = self.trello_request("POST", f"/1/boards", {"name": name, "idOrganization": self.organization_id})
```

Comment:
As mentioned above, nothing wrong with including organisation id but worth noting that it's optional - but saying that it all works smoothly if the user just skips the relevant environment variable so no problems with leaving it as is and letting users decide if they want to pass one